### ISSUE TITLE
reward shape

### ISSUE THREAD

**Issue by mehdimashayekhi** [mjacar/tensorflow-dcn-plus#3]
I have a question regarding the shape of the reward here https://github.com/mjacar/tensorflow-dcn-plus/blob/9358c85fb3b66e67d770b501655e89c62507a326/loss.py#L29. It is not really a big deal, but please correct me if I am wrong. ```final_logits```shape is ```batch_size*document_size*2``` and ```final_start_logits``` shape is ```batch_size*document_size```. According to the implementation, reward shape is ```batch_size*document_size``` . Shouldn't the reward shape be ```batch_size```? and there are four of those (since max_iter is 4)

**Comment by mjacar** [comment]
I think the reward should be shape `max_iter*batch_size` since you're sampling across each decoder timestep (`max_iter`) for each sample in the batch (`batch_size`).

**Comment by mjacar** [comment]
Perhaps more to your point, I'm not really sure what I'm doing with `logits_shape` or why I'm doing it. It doesn't really matter anyway since I'm overwriting everything with `f1_score - baseline` (which looks right to me) but it really seems as if `reward = [ANY_ARBITRARY_VALUE] * 4` would work just as well. Maybe there's some weird Tensorflow reason why I did that but I don't really remember and it isn't obvious to me at a glance.

**Comment by mehdimashayekhi** [comment]
Thanks for reply. I agree, and that what I was said above, I am just confused when you initialized reward as the shape of ```logits_shape``` (```tf.shape(final_start_logits)```), and as far as my understanding this has shape batch_size*document_size.

**Comment by mehdimashayekhi** [comment]
thanks, absolutely, it doesn't matter, because you overwrited it, I was just curious if there is any reason doing that:)
